lib/utils/httputil/httputil.go

```diff
@@ -304,10 +304,15 @@ func Send(method, rawurl string, options ...SendOption) (resp *http.Response, er
 				opts.retry.backoffMax)
 		}
 		resp, err = client.Do(req)
+
+		httpFallbackDisabled := opts.httpFallbackDisabled
+		if !httpFallbackDisabled && is5xxResponse(resp) {
```

Comment:
Just to confirm, it is not a 5xx response that is returned when we need to do the https -> http fallback? Assuming that is a 4xx response (but want to be sure).

Reply:
As discussed offline, nginx will return 495-497
http://nginx.org/en/docs/http/ngx_http_ssl_module.html